parser/src/ast/statement.rs

```diff
@@ -64,6 +64,67 @@ pub enum Statement {
     /// Declares a bus related constraint
     BusEnforce(ListComprehension),
 }
+
+#[derive(Clone, Spanned, Debug)]
+pub struct Match {
+    #[span]
+    pub span: SourceSpan,
+    pub match_arms: Vec<MatchArm>,
+}
+
+impl Match {
+    pub fn new(span: SourceSpan, match_arms: Vec<MatchArm>) -> Self {
+        Self { span, match_arms }
+    }
+}
+
+impl Eq for Match {}
+impl PartialEq for Match {
+    fn eq(&self, other: &Self) -> bool {
+        self.match_arms == other.match_arms
+    }
+}
+
+impl fmt::Display for Match {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "match: ")?;
+        for arm in self.match_arms.iter() {
+            write!(f, "{arm}")?;
+        }
+        Ok(())
+    }
+}
+
+#[derive(Clone, Spanned, Debug)]
```

Comment:
We do not need to implement `Eq` manually
```suggestion
#[derive(Clone, Spanned, Debug, Eq)]
```

Reply:
Thank you, removed the Eq implementations here and below